src/sentry/integrations/slack/utils/channel.py

```diff
@@ -228,8 +228,10 @@ def check_user_with_timeout(
             # TODO: This is a problem if we don't go through all the users and eventually run in to someone with duplicate display name
             if time.time() > time_to_quit:
                 return SlackChannelIdData(prefix=_prefix, channel_id=None, timed_out=True)
-    except SlackApiError:
+    except SlackApiError as e:
         _logger.exception("rule.slack.user_check_error", extra=logger_params)
+        if "ratelimited" in str(e):
+            raise ApiRateLimitedError("Slack rate limited") from e
```

Comment:
hmm i think this warrants more discussion. since we used to use the in-house client for everything, i think this was the default. but if we want callers to be able to handle the various SlackApiErrors appropriately maybe we want to always convert the SlackApiError into the equivalent ApiError?

Reply:
Given that `ApiRateLimitedError` extends `ApiError`, is that what this is already doing?

Reply:
i mean should we always catch SlackApiError and raise ApiError from the SlackApiError?

Reply:
similar to https://github.com/getsentry/sentry/pull/73666#discussion_r1664752164, maybe we should split the string by `\n` and check in the smaller string?